howdoi/howdoi.py

```diff
@@ -372,6 +378,29 @@ def _clear_cache():
     return cache.clear()
 
 
+def _parse_json(res, args):
+    """
+    @res: json object with answers and metadata
+    @args: command-line arguments (used for parsing)
+    returns: formatted string of text ready to be printed
+    """
```

Comment:
I like to keep the codebase pretty self explanatory without superfluous comments. While docstrings are nice, they're a bit out of place in this repo that I'm trying to keep succinct and readable. It's also a sign that we aren't naming our functions well.

I have a few suggestions:
1. Change `_get_instructions` to `_get_answers`.
Internally we already have the concept of `_get_answer`, so this will be a nice wrapper around that function to orchestrate the algorithm of fetching the question links and then the answers.

1. I like to use JSON like the bread in a sandwich. JSON can be read into a program, and JSON can be provided as output of a program, but the internal representation, at least for our uses, should be native objects within the language. As the code is now written, `_get_instructions` is returning JSON which must be `json.loads`'d and checked before it is returned to the user. Additionally, JSON is hard to typecheck and requires a lot of extra `json.dumps` all over the place. Let's have `_get_instructions` (now called `_get_answers`) return an array of answers.

1. Change the name of `_parse_json` to something like `_format_answers`. This function can take `answers` and `args` as parameters and convert either to a string for command line output or to JSON.

Reply:
Note: the JSON sandwich was inspired by the [Unicode sandwich](https://nedbatchelder.com/text/unipain/unipain.html), where you have various encodings that your program can accept and produce but everything inside the program is Unicode.
;)